src/collectors/jolokia/jolokia.py

```diff
@@ -119,53 +163,75 @@ def check_mbean(self, mbean):
             for chkbean in self.mbeans:
-                if re.match(chkbean, mbean) is not None or \
-                   re.match(chkbean, mbeanfix) is not None:
+                if chkbean.match(mbean) is not None or \
+                   chkbean.match(mbeanfix) is not None:
                     return True
         else:
             if mbean in self.mbeans or mbeanfix in self.mbeans:
                 return True
 
     def collect(self):
-        listing = self.list_request()
-        try:
-            domains = listing['value'] if listing['status'] == 200 else {}
-            for domain in domains.keys():
-                if domain not in self.IGNORE_DOMAINS:
-                    obj = self.read_request(domain)
+        if not self.domains:
+            self._get_domains()
+        for domain in self.domains:
+            if domain not in self.IGNORE_DOMAINS:
+                obj = self.read_request(domain)
+                try:
                     mbeans = obj['value'] if obj['status'] == 200 else {}
-                    for k, v in mbeans.iteritems():
-                        if self.check_mbean(k):
-                            self.collect_bean(k, v)
-        except KeyError:
-            # The reponse was totally empty, or not an expected format
-            self.log.error('Unable to retrieve MBean listing.')
+                except KeyError:
+                    # The reponse was totally empty, or not an expected format
+                    self.log.error('Unable to retrieve domain %s.', domain)
+                    continue
+                for k, v in mbeans.iteritems():
+                    if self.check_mbean(k):
+                        self.collect_bean(k, v)
 
     def read_json(self, request):
         json_str = request.read()
         return json.loads(json_str)
 
-    def list_request(self):
+    def _list_request(self):
+        """Retruns a dictionary with JMX domain names as keys"""
         try:
-            url = "http://%s:%s/%s%s" % (self.config['host'],
-                                         self.config['port'],
-                                         self.config['path'],
-                                         self.LIST_URL)
-            response = urllib2.urlopen(self._create_request(url))
-            return self.read_json(response)
-        except (urllib2.HTTPError, ValueError):
-            self.log.error('Unable to read JSON response.')
+            # https://jolokia.org/reference/html/protocol.html
+            # The optional parameter maxDepth can be used to restrict the depth
+            # of the return tree. Two value are possible: A maxDepth of 1
+            # restricts the return value to a map with the JMX domains as keys,
+            # a maxDepth of 2 truncates the map returned to the domain names
+            # (first level) and the MBean's properties (second level). The final
+            # values of the maps don't have any meaning and are dummy values.
```

Comment:
We can probably simplify this comment to explain why we've chosen maxDepth=1 and refer to the doc for the rest of the details/options.